doc/user/content/transform-data/troubleshooting.md

```diff
@@ -277,6 +277,317 @@ If you've gone through the dataflow troubleshooting and do not want to make any
+  [`mz_catalog.mz_objects`](/sql/system-catalog/mz_catalog/#mz_objects) to find
+  the actual name corresponding to the `global_id`.  The `WHERE mo.name IN ...`
+  clause of the query ensures we only see information about this index and view.
+  If you leave this `WHERE` clause out, you will see information on _every_
+  view, materialized view, and index on your current cluster.
+
+- The `operator` is indented using [`REPEAT`](/sql/functions/#repeat) and
+  `mz_lir_mapping.nesting`. The indenting, combined with ordering by
+  `mz_lir_mapping.lir_id` descending, gives us a tree-shaped view of the LIR
+  plan.
+
+### Attributing memory usage
+
+{{< tip >}}
+
+If you have not read about [attributing computation time](#attributing-computation-time), please do so first, as it explains some core concepts.
+
+{{< /tip >}}
+
+To find the memory usage of each operator for the index and view, join
+[`mz_introspection.mz_lir_mapping`](/sql/system-catalog/mz_introspection/#mz_lir_mapping)
+with
+[`mz_introspection.mz_arrangement_sizes`](/sql/system-catalog/mz_introspection/#mz_arrangement_sizes):
+
+```sql
+  SELECT mo.name AS name, global_id, lir_id, parent_lir_id, REPEAT(' ', nesting * 2) || operator AS operator,
+         pg_size_pretty(SUM(size)) AS size
+    FROM           mz_introspection.mz_lir_mapping mlm
+         LEFT JOIN mz_introspection.mz_arrangement_sizes mas
+                ON (mlm.operator_id_start <= mas.operator_id AND mas.operator_id < mlm.operator_id_end)
+              JOIN mz_catalog.mz_objects mo
+                ON (mlm.global_id = mo.id)
+   WHERE mo.name IN ('wins_by_item', 'winning_bids')
+GROUP BY mo.name, global_id, lir_id, operator, parent_lir_id, nesting
+ORDER BY global_id, lir_id DESC;
+```
+
+The query produces results similar to the following (your specific numbers will
+vary):
+
+{{< yaml-table data="query_attribution_memory_usage_output" >}}
+
+The results show:
+
+- The [`TopK`](/transform-data/idiomatic-materialize-sql/top-k/) is
+  overwhelmingly responsible for memory usage.
+
+- Arranging the `bids` table (`lir_id` 4) and `auctions` table (`lir_id` 2) are
+  cheap in comparison as is arranging the final output in the `wins_by_item`
+  index (`lir_id` 8).
+
+### Attributing `TopK` hints
+
+{{< tip >}}
+
+If you have not read about [attributing computation time](#attributing-computation-time), please do so first, as it explains some core concepts.
+
+{{< /tip >}}
+
+The
+[`mz_introspection.mz_expected_group_size_advice`](/sql/system-catalog/mz_introspection/#mz_expected_group_size_advice)
+looks at your running dataflow and suggests parameters you can set. We
+can attribute this to particular parts of our query using
+[`mz_introspection.mz_lir_mapping`](/sql/system-catalog/mz_introspection/#mz_lir_mapping):
+
+```sql
+  SELECT mo.name AS name, mlm.global_id AS global_id, lir_id, parent_lir_id, REPEAT(' ', nesting * 2) || operator AS operator,
+         levels, to_cut, hint, pg_size_pretty(savings)
```

Comment:
I moved the hint before pg_size_pretty ... so that the hint shows in the page.  I could also rename columns so that they all fit in ... but, eh.

Reply:
Honestly, there's something to be said for not even selecting out `levels` or `to_cut`, since they feel pretty inward looking.